### PR TITLE
Gh4018 option to not enforce M_HU_PI_Item_Product in DD_OrderLine

### PR DESCRIPTION
**Pull request by metas-rc**
https://github.com/metasfresh/metasfresh/issues/4018

### CODE REVIEW

File: de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/ddorder/spi/impl/DDOrderLineHUDocumentHandler.java [hunk 1]
```diff
@@ -42,6 +43,7 @@
  */
 public class DDOrderLineHUDocumentHandler implements IHUDocumentHandler
 {
+	private final static String SYSCONFIG_DD_OrderLine_Enforce_M_HU_PI_Item_Product = "DD_OrderLine_Enforce_M_HU_PI_Item_Product";
```
Comment (metas-ts) [inline de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/ddorder/spi/impl/DDOrderLineHUDocumentHandler.java]: please prepend de.metas.handlingunits

File: de.metas.handlingunits.base/src/main/sql/postgresql/system/5493530_sys_gh4018_SYSConfig_Enforce_DDOrderline_M_HU_PI_Item_Product.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+-- 2018-05-14T16:22:03.594
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+INSERT INTO AD_SysConfig (AD_Client_ID,AD_Org_ID,AD_SysConfig_ID,ConfigurationLevel,Created,CreatedBy,EntityType,IsActive,Name,Updated,UpdatedBy,Value) VALUES (0,0,541210,'S',TO_TIMESTAMP('2018-05-14 16:22:03','YYYY-MM-DD HH24:MI:SS'),100,'de.metas.handlingunits','Y','DD_OrderLine_Enforce_M_HU_PI_Item_Product',TO_TIMESTAMP('2018-05-14 16:22:03','YYYY-MM-DD HH24:MI:SS'),100,'-')
```
Comment (metas-ts) [inline de.metas.handlingunits.base/src/main/sql/postgresql/system/5493530_sys_gh4018_SYSConfig_Enforce_DDOrderline_M_HU_PI_Item_Product.sql]: * why did you add the `-` instead of Y or N? 
* please add some documentation about what the sysconfig's implications are. Both to this `AD_SysConfig` record and the issue. then flag the issue as `type:documentation`. In particular, please document how *this* sysconfig affects the actual issue around the CuQty

File: de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/ddorder/spi/impl/DDOrderLineHUDocumentHandler.java (hunk 1 above)
Comment (metas-rc) [reply de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/ddorder/spi/impl/DDOrderLineHUDocumentHandler.java]: done

**Comment by metas-rc** [comment]
I added a new sys config, named de.metas.handlingunits.DD_OrderLine_Enforce_M_HU_PI_Item_Product.

If this sys config is set on 'Y',  the DD_OrderLine will always have an M_HU_PI_Item_Product if fitting Packing Instructions are found in the db, even if the DD_Orderline was initially for CUs only. 
This M_HU_PI_Item_Product setting leads to also setting a qtyTU and the qtyCU will change to the qty that fitts those instructions (example: if a box fits 10 CUs, the qtyCU of the DD_OrderLine will be 10, no matter what it was before).

Sys config =  'N'  means that this functionality is not needed, and the DD_OrderLine will remain as it was created ( e.g. based on an inoutLine for CUs only)

File: de.metas.handlingunits.base/src/main/sql/postgresql/system/5493530_sys_gh4018_SYSConfig_Enforce_DDOrderline_M_HU_PI_Item_Product.sql (hunk 2 above)
Comment (metas-rc) [reply de.metas.handlingunits.base/src/main/sql/postgresql/system/5493530_sys_gh4018_SYSConfig_Enforce_DDOrderline_M_HU_PI_Item_Product.sql]: Done
